tests/pytest_tests/system_tests/test_artifacts/test_wandb_artifacts_full.py

```diff
@@ -214,6 +214,34 @@ def test_remove_after_log(wandb_init):
             retrieved.remove("file1.txt")
 
 
+def test_download_uses_cache(wandb_init, tmp_path, monkeypatch):
```

Comment:
Is this different from these other tests:

* `test_mutable_uploads_with_cache_enabled`
* `test_immutable_uploads_with_cache_enabled`

Also, these tests are *really* slow. This command took me 3.5 minutes!

```
pytest --durations=20 tests/pytest_tests/system_tests/test_artifacts/test_wandb_artifacts_full.py
```

This particular test takes ~4 seconds. Is there any way to make these tests faster? Why do they take so long to run?

Reply:
Or if you sync to main, a more accurate way to test would be

```
nox -s system_tests -- \
  tests/pytest_tests/system_tests/test_artifacts/test_wandb_artifacts_full.py
```

which will test both with and without wandb-core, and which will properly rebuild the Go code as needed. (You can also use it with `-db uv` :)

Reply:
Sort of trivially, those two tests both evaluate uploads, which this PR doesn't enable (the next one in the stack does though, and re-enables those tests). I'll work on the test timings, but let's file separate tickets for that.